matlab/CMakeLists.txt

```
@@ -17,6 +17,84 @@
 
 cmake_minimum_required(VERSION 3.20)
 
+# Build the Arrow C++ libraries WITHOUT bundled GoogleTest binaries.
+function(BUILD_ARROW)
```

Lowercase is better because we use lowercase for function call:

```suggestion
function(build_arrow)
```

---

Can we unify `build_arrow` and `build_arrow_and_gtest` by using function arguments?
We can use `cmake_parse_arguments()` for processing arguments: https://cmake.org/cmake/help/latest/command/cmake_parse_arguments.html
See also `resolve_dependency` macro that uses `cmake_parse_arguments()`: https://github.com/apache/arrow/blob/master/cpp/cmake_modules/ThirdpartyToolchain.cmake#L200-L204

---

Thanks - this is a really good idea!

We factored out the GoogleTest-specific logic into a macro called `build_gtest` for code cleanliness and added a `BUILD_GTEST` argument to `build_arrow` in order to enable us to remove the `build_arrow_and_gtest` function. `build_arrow` now just calls `build_gtest` when `BUILD_GTEST` is supplied.